### PR TITLE
Return LibraryCardInfo instead of BasicLibrary with KeepInfo

### PR DESCRIPTION
**Pull request by leogrim**
@jaredpetker @DerekBurch @ryanpbrewster @andrewconner

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryCardCommander.scala [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+      (info, viewerMem, subscriptions)
+    }
+  }
+
+  @StatsdTiming("libraryInfoCommander.createLibraryCardInfo")
+  private def createLibraryCardInfo(lib: Library, image: Option[LibraryImage], owner: BasicUser, numFollowers: Int,
+    followers: Seq[BasicUser], numCollaborators: Int, collaborators: Seq[BasicUser], isFollowing: Option[Boolean], membershipInfoOpt: Option[LibraryMembershipInfo], inviteInfoOpt: Option[LibraryInviteInfo], permissions: Set[LibraryPermission], path: String, orgView: Option[BasicOrganizationView]): LibraryCardInfo = {
+    LibraryCardInfo(
+      id = Library.publicId(lib.id.get),
+      name = Library.getDisplayName(lib.name, lib.kind),
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryCardCommander.scala:162]: Also changed this from `lib.name`

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryMembershipCommander.scala [hunk 2]
```diff
@@ -400,4 +405,21 @@ class LibraryMembershipCommanderImpl @Inject() (
...
+      db.readOnlyReplica { implicit s =>
+        val membershipOpt = libraryMembershipRepo.getWithLibraryIdAndUserId(libraryId, userId)
+        membershipOpt.map(createMembershipInfo)
+      }
+    }
+  }
+
+  def getLibrariesWithWriteAccess(userId: Id[User]): Set[Id[Library]] = {
+    db.readOnlyMaster { implicit session => libraryMembershipRepo.getLibrariesWithWriteAccess(userId) }
+  }
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryMembershipCommander.scala:424]: :+1:

File: kifi-backend/modules/shoebox/test/com/keepit/controllers/mobile/MobileKeepsControllerTest.scala [hunk 3]
```diff
@@ -61,6 +62,15 @@ class MobileKeepsControllerTest extends Specification with ShoeboxTestInjector w
...
   def prenormalize(url: String)(implicit injector: Injector): String = normalizationService.prenormalize(url).get
 
+  def libraryCard(libraryId: Id[Library])(implicit injector: Injector): LibraryCardInfo = {
+    val viewerOpt = inject[FakeUserActionsHelper].fixedUser.flatMap(_.id)
+    db.readOnlyMaster { implicit session =>
+      val library = libraryRepo.get(libraryId)
+      val owner = basicUserRepo.load(library.ownerId)
+      inject[LibraryCardCommander].createLibraryCardInfo(library, owner, viewerOpt, withFollowing = true, ProcessedImageSize.Medium.idealSize)
+    }
+  }
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/test/com/keepit/controllers/mobile/MobileKeepsControllerTest.scala:72]: Why doesn't the `LibraryCardCommander` just do this? This seems super pleasant.

Comment (leogrim) [inline kifi-backend/modules/shoebox/test/com/keepit/controllers/mobile/MobileKeepsControllerTest.scala:72]: It's only necessary in tests at this point

**Pull request merged**